### PR TITLE
Improve flexibility + adaptability of banner

### PR DESCRIPTION
**Pull request by antony**
Just opening a PR so that we can review changes / add comments etc :)

Will fix https://github.com/beyonk-adventures/gdpr-cookie-consent-banner/issues/2

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -1,9 +1,19 @@
+## GDPR Cookie Consent banner (fork)
```
Comment (antony) [inline README.MD]: We should clear this up before merge - I suggest a contributors section @mariojankovic or a contributors file, or both.

The content below can probably be put in a changelog and we can bump the major version.

File: README.MD [hunk 2]
```diff
@@ -45,11 +55,7 @@ then import the banner into your code:
 
 ##### From a CDN
 
-You can include the minified script directly from unpkg cdn:
-
-```html
-<script src="//unpkg.com/@beyonk/gdpr-cookie-consent-banner@3.1.0/dist/browser/bundle.min.js"></script>
-```
+No CDN link as of now. Please see the methods above and below on how to either include the esm package or a local file.
```
Comment (antony) [inline README.MD]: We can probably put the original line back here and remove the `version` from the URL

File: README.MD [hunk 3]
```diff
@@ -117,57 +123,56 @@ const options = {
...
+  closeLabel: 'Close window',
+
+  /**
+   * Better cookie descriptions
+   */
+  choices: {
+    necessary: {
+      label: 'Required cookies',
+      description:
+        "These can't be turned off as they are used to control all the other cookies",
```
Comment (antony) [inline README.MD]: This is great!

File: src/browser/main.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
-'use strict'
-
 import Banner from '../components/banner.svelte'
 
-function attachBanner (target, data = {}) {
+function attachBanner(target, props = {}) {
   const banner = new Banner({
     target,
-    data
+    props
```
Comment (antony) [inline src/browser/main.js]: ah yes - this is a bad port from v2 on my behalf, that would be why it wasn't working for James. I use it with Svelte so in my case, this `main.js` is never used.

File: src/components/banner.svelte [hunk 5]
```diff
@@ -231,28 +84,27 @@
     }
   })
 
-  function setCookie (choices) {
+  function setCookie(choices) {
```
Comment (antony) [inline src/components/banner.svelte]: I can't remember if I linted this or not, but the @beyonk/eslint-config should (I think) pick up that there needs to be a space between function name and brackets.

Not to worry though, I can just lint auto-fix it before merge.

Comment (mariojankovic) [reply src/components/banner.svelte]: Sorry, this seems to be my prettier config. I didn't bother to actually set it up with your eslint config so it did what it did. :) We could lint it properly if you want to.

File: src/browser/main.js (hunk 4 above)
Comment (mariojankovic) [reply src/browser/main.js]: Yes, it worked for me in Svelte too so I didn't really bother until I saw his issue. Not sure why I picked it up other than wanting to get rid of styles to be honest.

File: src/components/banner.svelte (hunk 5 above)
Comment (antony) [reply src/components/banner.svelte]: Yeah don't worry about it too much - I don't lint all my projects so there is a chance that it doesn't even have linting installed - I will rectify the linting post merge :)

File: README.MD (hunk 2 above)
Comment (mariojankovic) [reply README.MD]: Yes, sure. I'll revert that and remove the message on top as soon as this is merged.

File: README.MD (hunk 1 above)
Comment (mariojankovic) [reply README.MD]: Whatever you suggest. I'm not an opensource guy to be honest - I usually fork and adapt stuff to my needs, so if you want to let me know how you want to proceed on this and what I need to do that'd be great. I just didn't want to confuse people with the fork and the CDN being separate. So we'll revert to your readme and just update the options object.

Comment (mariojankovic) [reply README.MD]: I updated the readme to reflect what we discussed. Now there's just a few new lines in comparison to the original doc.

File: README.MD [hunk 6]
```diff
@@ -39,27 +39,31 @@ npm install --save-dev @beyonk/gdpr-cookie-consent-banner
...
+attachBanner(document.body)
 ```
 
 #### Using raw javascript
 
 ##### From a CDN
 
 ```html
-<script src="//unpkg.com/@beyonk/gdpr-cookie-consent-banner@3.1.0/dist/browser/bundle.min.js"></script>
+<link rel="stylesheet" href="//unpkg.com/@beyonk/gdpr-cookie-consent-banner@5.1.0/dist/style.css">
```
Comment (mariojankovic) [inline README.MD]: @antony I assumed you will bump the version to 5.1.0 after my PR has been merged, so I updated both CDN links to reflect that. I hope that's OK with you.

**Comment by antony** [comment]
Just found some time to try this out today. Looking fantastic!

**Pull request merged**